keras_cv/layers/preprocessing/random_translation.py

```diff
@@ -0,0 +1,255 @@
+        - *constant*: `(k k k k | a b c d | k k k k)` The input is extended by
+          filling all values beyond the edge with the same constant value k = 0.
+        - *wrap*: `(a b c d | a b c d | a b c d)` The input is extended by
+          wrapping around to the opposite edge.
+        - *nearest*: `(a a a a | a b c d | d d d d)` The input is extended by
+          the nearest pixel.
+      interpolation: Interpolation mode. Supported values: `"nearest"`,
+        `"bilinear"`.
+      seed: Integer. Used to create a random seed.
+      fill_value: a float represents the value to be filled outside the
+        boundaries when `fill_mode="constant"`.
+
+    Input shape:
+      3D (unbatched) or 4D (batched) tensor with shape:
+      `(..., height, width, channels)`,  in `"channels_last"` format.
+
+    Output shape:
+      3D (unbatched) or 4D (batched) tensor with shape:
+      `(..., height, width, channels)`,  in `"channels_last"` format.
+    """
+
+    def __init__(
+        self,
+        height_factor,
+        width_factor,
+        fill_mode="reflect",
+        interpolation="bilinear",
+        seed=None,
+        fill_value=0.0,
+        **kwargs,
+    ):
+        super().__init__(seed=seed, force_generator=True, **kwargs)
+        self.height_factor = height_factor
+        if isinstance(height_factor, (tuple, list)):
+            self.height_lower = height_factor[0]
+            self.height_upper = height_factor[1]
+        else:
+            self.height_lower = -height_factor
+            self.height_upper = height_factor
+        if self.height_upper < self.height_lower:
+            raise ValueError(
+                "`height_factor` cannot have upper bound less than "
+                f"lower bound, got {height_factor}"
+            )
+        if abs(self.height_lower) > 1.0 or abs(self.height_upper) > 1.0:
+            raise ValueError(
+                "`height_factor` must have values between [-1, 1], "
+                f"got {height_factor}"
+            )
+
+        self.width_factor = width_factor
+        if isinstance(width_factor, (tuple, list)):
+            self.width_lower = width_factor[0]
+            self.width_upper = width_factor[1]
+        else:
+            self.width_lower = -width_factor
+            self.width_upper = width_factor
+        if self.width_upper < self.width_lower:
+            raise ValueError(
+                "`width_factor` cannot have upper bound less than "
+                f"lower bound, got {width_factor}"
+            )
+        if abs(self.width_lower) > 1.0 or abs(self.width_upper) > 1.0:
+            raise ValueError(
+                "`width_factor` must have values between [-1, 1], "
+                f"got {width_factor}"
+            )
+
+        check_fill_mode_and_interpolation(fill_mode, interpolation)
+
+        self.fill_mode = fill_mode
+        self.fill_value = fill_value
+        self.interpolation = interpolation
+        self.seed = seed
+
+    @tf.function
```

Comment:
Isn't this already propagated by the user at model compilation arg `run_eagerly`?

Reply:
I assume this is intended for standalone usage -- but unsure if it's helpful since standalone usage with happen in tf.data, which is already graph based

Reply:
Ok, but as `run_eagerly` is by default `False` we have lost the control of this function with the user API.

We could recover it only with the "global"  `tf.config.run_functions_eagerly(True)` or editing the library.

But I don't know if it is something we care about or not.